lib/core/shared/sdk/funnelProtocol.js

```diff
@@ -64,13 +69,34 @@ class FunnelProtocol extends KuzzleEventEmitter {
           id: await global.kuzzle.ask('core:network:internal:connectionId:get'),
           protocol: this.id
         },
-        user: this.user
+        user: null
       };
     }
 
+    // Use the provided request.kuid, otherwise use the impersonatedId previously given at funnel construction
+    if (request.__kuid__) {
+      await this._impersonate(request.__kuid__);
+    }
+    else { // Reset user if impersonation is not required
+      this.requestOptions.user = null;
+    }
```

Comment:
Since we already have the `kuid` I'm not sure we need to get the entire user object the the `kuid` is enough for the rights and metadata

Reply:
It seems to be insufficient, since the FunnelProtocol needs it. And IIRC there is only 1 FunnelProtocol for all impersonated SDKs: loading the User object in an async task means that race conditions can (and probably will) occur when processing concurrent requests.
Or did I miss something?

Reply:
Seems like the only place where the user is required, it actually uses only the **_id** in it. (KuzzleContext)

Reply:
Ok, so. Now we store the user and the required request options in the local context instead, this should solve our race condition issue. Can you guys confirm this? @scottinet @Aschen